docs/en/operations/settings/settings.md

```diff
@@ -404,6 +404,32 @@ Possible values:
 
 Default value: 0.
 
+## any_join_distinct_right_table_keys {#any_join_distinct_right_table_keys}
+
+Enables or disables specific ClickHouse server behavior in `ANY INNER|LEFT|RIGHT| JOIN` operations.
+
+When the behavior enabled:
```

Comment:
Could we add `(old logic)` at the end of this line?

Reply:
"old/new" is very strange in docs. We need to write text about current state. We can describe current default behavior and place a link there.

Reply:
Old means legacy. We must document which bahaviour is legacy and would be removed later.

Reply:
we can explicitly tell it is deprecated. We haven't switched the default yet? Otherwise it is highly informative to write version when this beahvior was changed in default configuration.

Reply:
This setting enables legacy bahaviour. Setting itself is not deprecated. It's backward compatibility setting. We just should mark that 0 means OK and 1 means not OK - legacy. For now it's not clear and user can think that it's normal to set it to 1. But it's not.